### DISCUSSION TITLE
I cannot use function crypto_core_ed25519_random in iOS, but I can use other functions like sodium_init

### DISCUSSION THREAD

**Discussion by wangfenghappy** [jedisct1/libsodium#1380] [Q&A]
#import "sodium.h"
#import "crypto_core_ed25519.h"

static void ecdh(void){
    int res = sodium_init();
    if (res != -1){
        NSLog(@"sodium_init success!");
    }
//    
    unsigned char * kd = calloc(32, sizeof(unsigned char));
    crypto_core_ed25519_random(kd);
    NSLog(@"sodium_init %s", kd);
}

Undefined symbols for architecture x86_64:
  "_crypto_core_ed25519_random", referenced from:

**Comment by jedisct1** [comment]
> `#import "crypto_core_ed25519.h"`

You should just import `sodium.h`

How did you build the library? Is it a full build or a minimal build?
